stdlib/format.mli

```diff
@@ -236,6 +236,11 @@ val pp_print_string : formatter -> string -> unit
 val print_string : string -> unit
 (** [pp_print_string ppf s] prints [s] in the current pretty-printing box. *)
 
+val pp_with_size : formatter -> int -> unit
+val with_size: int -> unit
+(** [pp_with_size ppf w] sets the width of the next element to be printed to
+    [w]. *)
```

Comment:
I think it would help to clarify that `pp_with_size` does not affect the apparent size of the next element if there are any intervening box opens, box closes, or break hints that break. I think that to understand this function requires users to think not in terms of a tree of boxes, but to consider its fringe as a linear sequence of "elements" (that end up being strings). It is not clear to me that this will be obvious.

Reply:
I agree that some clarification is required, what about

>  [pp_with_size ppf w] sets the width of the next element to be printed on the current line to [w], this does not affect the width of boxes nor the width of elements printed after a linebreak.

?

Reply:
I think that still leaves the possibility for people to expect that `@<10>@[%s` will affect the size of the string given for the `%s`. So perhaps:

> [pp_with_size ppf w] sets the width of the next element to be printed on the current line to [w]; this does not have any effect if there are any line breaks, tabulation breaks, or any boxes are opened or closed before the next element.